clients/ts/signalr/tests/Common.ts

```diff
@@ -21,3 +22,38 @@ export function eachEndpointUrl(action: (givenUrl: string, expectedUrl: string)
 
     urls.forEach((t) => action(t[0], t[1]));
 }
+
+type errorFn = (error: string) => boolean;
+
+export class VerifyLogger implements ILogger {
+    public unexpectedErrors: string[];
+    private expectedErrors: errorFn[];
+
+    public constructor(...expectedErrors: Array<RegExp | string | errorFn>) {
+        this.unexpectedErrors = [];
+        this.expectedErrors = [];
+        expectedErrors.forEach((element) => {
+            if (element instanceof RegExp) {
+                this.expectedErrors.push((e) => element.test(e));
+            } else if (typeof element === "string") {
+                this.expectedErrors.push((e) => element === e);
+            } else {
+                this.expectedErrors.push(element);
+            }
+        }, this);
+    }
+
+    public static async run(fn: (logger: VerifyLogger) => Promise<any>, ...expectedErrors: Array<RegExp | string | errorFn>): Promise<any> {
+        const logger = new VerifyLogger(...expectedErrors);
+        await fn(logger);
+        expect(logger.unexpectedErrors.length).toBe(0);
+    }
+
+    public log(logLevel: LogLevel, message: string): void {
+        if (logLevel >= LogLevel.Error) {
+            if (this.expectedErrors.filter((fn) => fn(message)).length === 0) {
```

Comment:
`.some` is `.Any` in JavaScript (supported down to IE 9, though these tests run in Node anyway)